source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,293 @@
+===================
+Compound Operations
+===================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+On this page you will learn how to perform **compound operations** with the
+MongoDB Java driver. 
+
+Compound operations consist of a read and write operation performed as one
+**atomic operation**. An atomic operation is an irreducible series of one or
+more database operations such that all occur or none occur.
+
+Atomic operations can help avoid certain kinds of **race conditions** in your code. A
+race condition occurs when your code's behavior is dependent on the order of
+uncontrollable events.
+
+MongoDB supports the following compound operations:
+
+- Find and update one document
+- Find and replace one document
+- Find and delete one document
+
+If you need to perform more complex tasks atomically, such as reading and
+writing to more than one document, use **transactions**.
+
+Transactions are a feature of MongoDB and other databases that lets you define
+an arbitrary sequence of database commands as an atomic operation.  
+
+For more information on atomic operations and atomicity, see 
+:manual:`the MongoDB manual entry for atomicity and transactions </core/write-operations-atomicity/>`.
+
+For more information on transactions, 
+:manual:`see the MongoDB manual entry for transactions </core/transactions/>`.
+
+How to Use Compound Operations
+------------------------------
+
+This section shows how to use each compound operation with the MongoDB Java Driver.
+
+The following examples use a collection containing these two sample documents.
+
+.. code-block:: json
+
+    {"_id": 1, "food": "donut", "color": "green"}
+    {"_id": 2, "food": "pear", "color": "yellow"}
```

Comment:
Thanks for the note. The json formatting on the page is meant to mimic the default behavior of the Java Driver's `BsonDocument.toJson()` method which does not include spaces. Sample output from driver:

> `{"food": "donut", "color": "green"}`

Because of this, I would rather not include spaces.